app/modules/audioplayer/components/controls/controls.component.ts

```diff
@@ -12,12 +12,106 @@ import {PlayQueueItem} from '../../models/play_queue_item.model';
 })
 
 export class PlayerControlsComponent {
-  private audio: any;
+
+  private audio: HTMLAudioElement;
   private playQueue: PlayQueue = PlayQueue.getInstance();
 
+  private timeTick: string;
+  private duration: string;
+
+  private timeTickPosition: string;
+  private progressBarWidth: number;
+
+  @ViewChild('audioPlayerProgressBarLine') audioPlayerProgressBarLine: ElementRef;
+  @ViewChild('audioPlayerHandle') audioPlayerProgressHandle: ElementRef;
+  @ViewChild('audioPlayerProgressBar') audioPlayerProgressBar: ElementRef;
+
   constructor() {
     this.audio = new Audio();
     this.playQueue.on('change:status', this.reactOnStatusChange, this);
+    this.timeTick = this.formatToHHMMSS(0);
+    this.duration = this.formatToHHMMSS(0);
+
+    this.timeTickPosition = '0px';
+
+    this.audio.addEventListener('canplay', () => {
+      this.duration = this.formatToHHMMSS(this.audio.duration);
```

Comment:
I would prefer to call a method instead of a inline function. Maybe you want to add logic later. It's nicer to keep the constructor clean